vtex-apps/glovo-integration
# Update order status

#### What problem is this solving?



This PR adds an event handler that receives status updates from the vtex.orders-broadcast app. It can be configured to listen to all or just some of the order status changes.
Right now it is configured to listen only when the order status changes to 'ready-for-handling' and 'invoiced' as can be seen  [here.](https://github.com/vtex-apps/glovo-integration/compare/update-order-status?expand=1#diff-d5dcd0c6aea7de2c8347424d20046555c29f8300a49b886d8416344ad4577a4dR14-R18)

#### How to test it?



To test it you can create a new order on [this Workspace](https://glovoordersfeed--powerplanet.myvtex.com) or create the invoice for existing orders in the store's [admin panel](https://glovoordersfeed--powerplanet.myvtex.com/admin/checkout/#/orders).
After creating a new order or create the invoice for an existing one you should get the POST on [Beeceptor.](https://beeceptor.com/console/testaffiliate)

#### Example of POST received on Beeceptor



```json
{
  "domain": "Marketplace",
  "orderId": "1123373285752-01",
  "currentState": "invoiced",
  "currentChangeDate": "2021-04-08T09:26:41.0132941Z"
}
```
